configure.ac

```diff
@@ -942,14 +950,12 @@
       [dnl When bootstrapping from the git submodule (flexdll directory), just
        dnl use that, however if another directory has been specified with
        dnl --with-flexdll=<path> then copy the contents of <path> to
-       dnl flexdll-sources. This allows us to guarantee that flexdll is always
-       dnl built at one directory level below the root, so we can specify
-       dnl ROOTDIR=..
+       dnl flexdll-sources.
       AS_IF([m4_normalize([test x"$with_flexdll" = 'x'
                             || test x"$with_flexdll" = 'xflexdll'])],
         [AS_IF([test -f 'flexdll/flexdll.h'],
           [flexdll_source_dir=flexdll
-          iflexdir='$(ROOTDIR)/flexdll'
+          iflexdir="$ocamlsrcdir\flexdll"
```

Comment:
```suggestion
iflexdir="$ocamlsrcdir\\flexdll"
```

Reply:
This mirrors similar code for winpthreads:
https://github.com/ocaml/ocaml/blob/e719a1f78c825c1fff4c10c2c47e044149da4caf/configure.ac#L1086

Reply:
:smile:
It flashed in my eyes because there is a version with `\\` just in the next block.
And indeed, the same inconsistency appears in the winpthreads block, shamefully :see_no_evil:

Reply:
Sorry to be bringing this in late, but this introduces an unescaped and absolute path into `OC_CPPFLAGS` in `Makefile.build_config.in` which is dangerous (in particular, building in `C:\Devel\ocaml-$(false)` now needs two levels of escaping, but works "correctly" on trunk 😁). If we've ended up with winpthreads also putting an absolute path into the build, really the change ought to be to make it equivalent here (i.e. `$(ROOTDIR)/winpthreads`, if necessary)

Can I check I understand correctly what the clang-cl part of the problem is - we obviously write nonsense into `ocamltest/ocamltest_config.ml` at the moment... is that actually causing an error?

Neither of those two flags should have made it ocamltest's cppflags, as they're specific CPPFLAGS for compiling the runtime, and that feels like a safer fix?

Reply:
Oo, sorry about that.

> we obviously write nonsense into ocamltest/ocamltest_config.ml at the moment... is that actually causing an error?

That was why I went with inlining the path for winpthreads in the MSVC-restoration work, indeed, since quite a few tests were failing when they could not find `pthread.h`. As I hadn’t understood the danger of not using `$(ROOTDIR)`, I didn’t revert to that version after the tests were fixed to use the proper API.

So I just checked that, with the following patch, the MSVC port compiles and tests fine (tested on `trunk`):

```diff
diff --git a/configure.ac b/configure.ac
index fbd8c2e833..53765c9508 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1083,7 +1083,7 @@ AS_IF([test x"$with_winpthreads_msvc" = "xno"],
                           || test x"$with_winpthreads_msvc" = x'winpthreads'])],
       [AS_IF([test -f 'winpthreads/src/winpthread_internal.h'],
         [winpthreads_source_dir=winpthreads
-        iwinpthreadsdir="$ocamlsrcdir\winpthreads\include"],
+        iwinpthreadsdir='winpthreads\include'],
         [AC_MSG_RESULT([required but not available (uninitialized submodule?)])
         AC_MSG_ERROR([exiting])])],
       [rm -rf winpthreads-sources
@@ -1093,7 +1093,7 @@ AS_IF([test x"$with_winpthreads_msvc" = "xno"],
         cp "$with_winpthreads_msvc"/src/*.h winpthreads-sources/src
         cp "$with_winpthreads_msvc"/include/*.h winpthreads-sources/include
         winpthreads_source_dir='winpthreads-sources'
-        iwinpthreadsdir="$ocamlsrcdir\\winpthreads-sources\\include"
+        iwinpthreadsdir='winpthreads-sources\include'
         winpthreadsmsg=" (from $with_winpthreads_msvc)"],
         [AC_MSG_RESULT([requested but not available])
         AC_MSG_ERROR([exiting])])])
@@ -1101,6 +1101,7 @@ AS_IF([test x"$with_winpthreads_msvc" = "xno"],
       [AC_MSG_RESULT([no])],
       [AC_MSG_RESULT([$winpthreads_source_dir$winpthreadmsg])
       OCAML_TEST_WINPTHREADS_PTHREAD_H([$iwinpthreadsdir])
+      iwinpthreadsdir='$(ROOTDIR)\'"$iwinpthreadsdir"
       internal_cppflags="$internal_cppflags -I$iwinpthreadsdir"])],
     [AS_IF([test x"$with_winpthreads_msvc" != 'x'],
       [AC_MSG_RESULT([requested but not supported])
```

> Neither of those two flags should have made it ocamltest's cppflags, as they're specific CPPFLAGS for compiling the runtime, and that feels like a safer fix?

Indeed, that would be much nicer!